hudi-spark-datasource/hudi-spark3-common/src/main/scala/org/apache/spark/sql/hudi/catalog/HoodieCatalog.scala

```diff
@@ -153,6 +156,77 @@ class HoodieCatalog extends DelegatingCatalogExtension
     }
   }
 
+  override def loadTable(ident: Identifier, timestamp: Long): Table = {
+    super.loadTable(ident) match {
+      case V1Table(catalogTable0) if sparkAdapter.isHoodieTable(catalogTable0) =>
+
+        // spark passes microseconds
+        val milliseconds = TimeUnit.MICROSECONDS.toMillis(timestamp)
+        val metaClient: HoodieTableMetaClient = HoodieTableMetaClient.builder()
+          .setBasePath(catalogTable0.location.toString)
+          .setConf(HadoopFSUtils.getStorageConf(SparkSession.active.sessionState.newHadoopConf))
+          .build()
+        HoodieInstantTimeGenerator.setCommitTimeZone(metaClient.getTableConfig.getTimelineTimezone)
+        val targetTimestamp = HoodieInstantTimeGenerator.formatDateWitTimeZone(new Date(milliseconds))
+
+        constructTable(ident, catalogTable0, targetTimestamp)
+
+      case t => t
+    }
+  }
+
+  override def loadTable(ident: Identifier, version: String): Table = {
+    super.loadTable(ident) match {
+      case V1Table(catalogTable0) if sparkAdapter.isHoodieTable(catalogTable0) =>
+
+        if (!HoodieInstantTimeGenerator.isValidInstantTime(version)) {
+          throw new AnalysisException(s"invalid snapshot id: $version")
+        }
+
+        constructTable(ident, catalogTable0, version);
+
+      case t => t
+    }
+  }
+
+  private def constructTable(ident: Identifier,
+                             catalogTable0: CatalogTable,
+                             timestampAsOf: String): Table = {
+    val catalogTable = catalogTable0.comment match {
+      case Some(v) =>
+        val newProps = catalogTable0.storage.properties + (
+          TableCatalog.PROP_COMMENT -> v,
+          DataSourceReadOptions.TIME_TRAVEL_AS_OF_INSTANT.key -> timestampAsOf
```

Comment:
Are only the comment and version timestamp missing here? What is the criteria to know which properties need to be supplemented?

Reply:
Here, we pass the parameters parsed during the process through the load method of TableCatalog, rather than in the table config or session parameters, other will construct when create relation. Since it is still a V1Table, the parameters can only be set in the table storage.